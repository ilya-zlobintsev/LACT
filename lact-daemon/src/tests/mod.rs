use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
