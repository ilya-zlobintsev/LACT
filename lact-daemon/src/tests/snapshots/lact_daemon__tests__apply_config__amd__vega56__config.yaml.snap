---
source: lact-daemon/src/tests/mod.rs
expression: write_commands
---
[
    "card0/device/pp_od_clk_voltage: r\n",
    "card0/device/pp_od_clk_voltage: s 0 852 800\n",
    "card0/device/pp_od_clk_voltage: s 1 991 900\n",
    "card0/device/pp_od_clk_voltage: s 2 1138 950\n",
    "card0/device/pp_od_clk_voltage: s 3 1269 1000\n",
    "card0/device/pp_od_clk_voltage: s 4 1312 1050\n",
    "card0/device/pp_od_clk_voltage: s 5 1474 1100\n",
    "card0/device/pp_od_clk_voltage: s 6 1538 1150\n",
    "card0/device/pp_od_clk_voltage: s 7 1630 1200\n",
    "card0/device/pp_od_clk_voltage: m 0 167 800\n",
    "card0/device/pp_od_clk_voltage: m 1 500 800\n",
    "card0/device/pp_od_clk_voltage: m 2 700 900\n",
    "card0/device/pp_od_clk_voltage: m 3 920 950\n",
    "card0/device/power_dpm_force_performance_level: auto",
    "card0/device/pp_od_clk_voltage: c\n",
]
